### DISCUSSION TITLE
slightly optimized version of the default JSONStorage

### DISCUSSION THREAD

**Discussion by MrPigss** [msiemens/tinydb#456] [Extensions]
[BetterJSONStorage](https://github.com/MrPigss/BetterJSONStorage) is a storage class that aims to improve the default JSONStorage.
It does that in 3 different ways:

- Compression
- Better parsing
- Threading

It uses BLOSC for compression so it results in filesizes a lot smaller than uncompressed db files.
These smaller filesizes also result in less diskIO so less time waiting.

Orjson is used to speed up  the parsing process . For more info checkout the orjson github and see how much faster it is compared to builtin json module. Also orjson has some extra options for datetime parsing and keys, wich looks usefull together with TinyDBTimestamps.

The way writing and reading is handled is also fundamentaly different from the default JSONStorage. Reading is always done from memory so it's lightning fast! All wrinting is also handled by a seperate thread so there aren't any slowdowns.

All these optimizations result in speeds almost 50x faster.
If you don't believe it, everything can be found on the github.

Let me know what you think!

**Comment by msiemens** [comment]
Great work @MrPigss! I added your project to TinyDB's [extension list](https://tinydb.readthedocs.io/en/latest/extensions.html) 🙂
